Source/DafnyCore/Generic/ReporterExtensions.cs

```diff
@@ -36,26 +40,27 @@ public static class ErrorReporterExtensions {
 
   private const string RelatedLocationCategory = "Related location";
   public const string RelatedLocationMessage = RelatedLocationCategory;
+  private const string RelatedMessageCategory = "Related message";
   public static readonly string PostConditionFailingMessage = new ProofObligationDescription.EnsuresDescription(null, null).FailureDescription;
   private static string FormatRelated(string related) {
     return $"Could not prove: {related}";
   }
 
-  public static IEnumerable<DafnyRelatedInformation> CreateDiagnosticRelatedInformationFor(IToken token, string? message) {
+  public static IEnumerable<DafnyRelatedInformation> CreateDiagnosticRelatedInformationFor(IToken token, string? message, bool usingSnippets) {
     var (tokenForMessage, inner, newMessage) = token is NestedToken nestedToken ? (nestedToken.Outer, nestedToken.Inner, nestedToken.Message) : (token, null, null);
     var dafnyToken = BoogieGenerator.ToDafnyToken(true, tokenForMessage);
-    if (dafnyToken is RangeToken rangeToken) {
+    if (!usingSnippets && dafnyToken is RangeToken rangeToken) {
```

Comment:
Isn't it still useful to get the related information even when using snippets? Or are you just maintaining the current behaviour here?

Reply:
This block will append some source code to the error message, similar to what snippets do but less pretty.